homeassistant/components/tesla_fleet/media_player.py

```diff
@@ -0,0 +1,149 @@
+"""Media player platform for Tesla Fleet integration."""
+
+from __future__ import annotations
+
+from tesla_fleet_api.const import Scope
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TeslaFleetConfigEntry
+from .entity import TeslaFleetVehicleEntity
+from .helpers import handle_vehicle_command
+from .models import TeslaFleetVehicleData
+
+STATES = {
+    "Playing": MediaPlayerState.PLAYING,
+    "Paused": MediaPlayerState.PAUSED,
+    "Stopped": MediaPlayerState.IDLE,
+    "Off": MediaPlayerState.OFF,
+}
+VOLUME_MAX = 11.0
+VOLUME_STEP = 1.0 / 3
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TeslaFleetConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Tesla Fleet Media platform from a config entry."""
+
+    async_add_entities(
+        TeslaFleetMediaEntity(vehicle, Scope.VEHICLE_CMDS in entry.runtime_data.scopes)
+        for vehicle in entry.runtime_data.vehicles
+    )
+
+
+class TeslaFleetMediaEntity(TeslaFleetVehicleEntity, MediaPlayerEntity):
+    """Vehicle media player class."""
+
+    _attr_device_class = MediaPlayerDeviceClass.SPEAKER
+    _attr_supported_features = (
+        MediaPlayerEntityFeature.NEXT_TRACK
+        | MediaPlayerEntityFeature.PAUSE
+        | MediaPlayerEntityFeature.PLAY
+        | MediaPlayerEntityFeature.PREVIOUS_TRACK
+        | MediaPlayerEntityFeature.VOLUME_SET
+    )
+    _volume_max: float = VOLUME_MAX
+
+    def __init__(
+        self,
+        data: TeslaFleetVehicleData,
+        scoped: bool,
+    ) -> None:
+        """Initialize the media player entity."""
+        super().__init__(data, "media")
+        self.scoped = scoped
+        if not scoped and data.signing:
+            self._attr_supported_features = MediaPlayerEntityFeature(0)
+
+    def _async_update_attrs(self) -> None:
+        """Update entity attributes."""
+        self._volume_max = (
+            self.get("vehicle_state_media_info_audio_volume_max") or VOLUME_MAX
+        )
+        self._attr_state = STATES.get(
+            self.get("vehicle_state_media_info_media_playback_status") or "Off",
```

Comment:
Why is Off a state?

Reply:
Well it is a state, but I am not sure why I didnt use the ENUM
`MediaPlayerState.OFF`

Reply:
Oh thats right "Off" is a key for `STATES` which translates Tesla's values to the ENUM

Reply:
Ah check, i missed that one as well